### DISCUSSION TITLE
Extending SQLLineage to understand functions as source in SQL

### DISCUSSION THREAD

**Discussion by sebastianheibo** [reata/sqllineage#653] [Q&A]
In the database software my company uses, a customized version of spark sql is used. I therefore have some SQL queries that have functions in the source of the query. 

An example query I have is:

SELECT
    column1,
    column2,
    etc.
FROM
    data_model("model_name", "table_name", "model_version")

How can I extend the functionality of the LineageRunner or other parts of the SQLLineage library to be able to parse the data_model() function?

Thank you!

**Comment by maoxingda** [comment]
针对于你这个用例，目前的行为是什么？

**Reply by sebastianheibo** [reply]
With the mentioned example I get something like this:
![image](https://github.com/user-attachments/assets/8ede10dd-5b4c-41b0-8474-5f73e497fd2d)
